source/parse.h

```diff
@@ -4648,7 +4648,18 @@ class parser
             error("Cpp2 is currently exploring the path of not allowing default arguments - use overloading instead", false);
             return {};
         }
-
+        if (is_named && is_returns) {
+            auto tok = n->name();
+            assert(tok);
+            if (tok->type() != lexeme::Identifier) {
+                error("expected identifier, not '" + tok->to_string(true) + "'",
+                    false, tok->position());
+            }
+            else if (!n->declaration->is_type()) {
```

Comment:
Sorry, I gave an incomplete feedback on this line the first time: I think the right query is not "is this declaration a type" (a parameter shouldn't ever be a type, it should be an object) but rather "does this declaration have a concrete type"... so I think what you want is `else if (n->declaration->has_wildcard_type())`.

Did you run regressions? I think the current condition above would reject existing valid named return cases?

Reply:
Sorry, last night I checked the error tests but I forgot to check the correct named return test. Now fixed.